lib/3scale/backend/worker.rb

```diff
@@ -19,22 +19,29 @@ module Worker
 
       DEFAULT_MAX_CONCURRENT_JOBS = 20
 
+      if configuration.worker_prometheus_metrics.enabled
+        require '3scale/backend/worker_metrics'
+      end
+
+      # Conditional require is done to require async-* libs only when
+      # needed and avoid possible side-effects.
+      if configuration.redis.async
+        require '3scale/backend/worker_async'
+      else
+        require '3scale/backend/worker_sync'
+      end
+
       def self.new(options = {})
         Logging::Worker.configure_logging(self, options[:log_file])
         Logging::External.setup_worker
 
         if configuration.worker_prometheus_metrics.enabled
-          require '3scale/backend/worker_metrics'
           WorkerMetrics.start_metrics_server
         end
 
-        if options[:async]
-          # Conditional require is done to require async-* libs only when
-          # needed and avoid possible side-effects.
-          require '3scale/backend/worker_async'
+        if configuration.redis.async
           WorkerAsync.new(options)
```

Comment:
I like the changes. In general we should have a factory class or a service to get us an instance but no need to complicate something used at only one place.